Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -256,6 +269,175 @@ public boolean logEvent(Context context) {
+            @Override
+            public void onBillingSetupFinished(BillingResult billingResult) {
+                if (billingResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+
+
+                    billingClient.queryProductDetailsAsync(querySubsProductDetailsParams, (billingQueryResult, productDetailsList) -> {
+                        if (billingQueryResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+                            List<BranchUniversalObject> buos = new ArrayList<>();
+                            CurrencyType currency = null;
+                            double revenue = 0.00;
+                            String purchasedProductType = "";
+
+                            for (ProductDetails product : productDetailsList) {
+                                double price;
+                                int quantity;
+
+                                purchasedProductType = "Subscription";
+                                currency = CurrencyType.valueOf(product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceCurrencyCode());
+                                price = product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceAmountMicros() / 1_000_000;
+                                quantity = 1;
+                                revenue += price * quantity;
+
+
+                                BranchUniversalObject buo = new BranchUniversalObject()
+                                        .setCanonicalIdentifier(product.getProductId())
+                                        .setTitle(product.getTitle())
+                                        .setContentMetadata(
+                                                new ContentMetadata()
+                                                        .addCustomMetadata("product_type", product.getProductType())
+                                                        .setPrice(price, currency)
+                                                        .setProductName(product.getName())
+                                                        .setQuantity((double) quantity)
+                                                        .setContentSchema(BranchContentSchema.COMMERCE_PRODUCT));
+                                buos.add(buo);
+                            }
+
+                            new BranchEvent(BRANCH_STANDARD_EVENT.PURCHASE)
+                                    .setCurrency(currency)
+                                    .setDescription(purchase.getOrderId())
+                                    .setCustomerEventAlias(purchasedProductType)
+                                    .setRevenue(revenue)
+                                    .addCustomDataProperty("package_name", purchase.getPackageName())
+                                    .addCustomDataProperty("order_id", purchase.getOrderId())
```

Comment:
There are quite a few hardcoded strings here, are there constants you can use from the library?